### PR TITLE
Apply configured uuid representation to DB#command encoding and decoding

### PR DESCRIPTION
**Pull request by jyemin**
JAVA-4450

### CODE REVIEW

File: driver-legacy/src/main/com/mongodb/DB.java [hunk 1]
```diff
@@ -84,7 +83,7 @@
         this.name = name;
         this.executor = executor;
         this.collectionCache = new ConcurrentHashMap<String, DBCollection>();
-        this.commandCodec = new DBObjectCodec(getDefaultCodecRegistry());
+        this.commandCodec = new DBObjectCodec(mongo.getCodecRegistry());
```
Comment (jyemin) [inline driver-legacy/src/main/com/mongodb/DB.java:86]: This fixes the encoding

File: driver-legacy/src/main/com/mongodb/DB.java [hunk 2]
```diff
@@ -522,7 +521,7 @@ public String toString() {
     CommandResult executeCommand(final BsonDocument commandDocument, final ReadPreference readPreference) {
         return new CommandResult(executor.execute(new CommandReadOperation<BsonDocument>(getName(), commandDocument,
                                                                                          new BsonDocumentCodec()),
-                                                  readPreference, getReadConcern()));
+                                                  readPreference, getReadConcern()), getDefaultDBObjectCodec());
```
Comment (jyemin) [inline driver-legacy/src/main/com/mongodb/DB.java:524]: And this fixes the decoding

File: driver-legacy/src/main/com/mongodb/DB.java [hunk 3]
```diff
@@ -566,6 +565,13 @@ ReadPreference getCommandReadPreference(final DBObject command, @Nullable final
         }
     }
 
+    Codec<DBObject> getDefaultDBObjectCodec() {
```
Comment (jyemin) [inline driver-legacy/src/main/com/mongodb/DB.java:568]: Basically copied from DBCollection, but it can't be exactly the same so not really shareable.

**Review by rozza — APPROVED** [review]
:+1: lgtm

**Pull request merged**